modules/apps/frontend-js/frontend-js-item-selector-web/src/main/resources/META-INF/resources/item_selector/DetachedItemSelectorModal.tsx

```diff
@@ -0,0 +1,46 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {useModal} from '@clayui/modal';
+import React, {useEffect} from 'react';
+
+import {IItemSelectorModalProps} from './ItemSelectorModal';
+import ItemSelectorModalContent from './ItemSelectorModalContent';
+
+export type TDetachedItemSelectorModal<T> = Omit<
+	IItemSelectorModalProps<T>,
+	'observer' | 'onOpenChange' | 'open'
+>;
+
+function DetachedItemSelectorModal<T extends Record<string, any>>(
+	props: TDetachedItemSelectorModal<T>
+) {
+	const {observer, onOpenChange, open} = useModal();
```

Comment:
The key difference in new modal variant is internal `useModal` hook. I not sure if there are any benefits in having `observer, onOpenChange, open` as controlled props. In theory, the whole solution could be refactored to this, it is something you can consider for future. For this PR, I didn't want to break existing API.